app/src/main/java/com/example/drawingApp/SelectedColorView.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+
+@SuppressLint("ClickableViewAccessibility")
+/**
+ * This class is ued to create a custom view
+ * the view shows a solid circle
+ * when clicked it will cycle forward in its color order
+ * the color order is Black, Red, Green, Blue
+ *
+ */
+class SelectedColorView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    private var states = arrayOf(Color.BLACK,Color.RED,Color.GREEN,Color.BLUE)
+    private var state = states[0]
+    private val paint = Paint()
+    var passingColorCircle: (() -> Unit)? = null
```

Comment:
I don't understand what this does by the name. Passing? `onColorChanged`? What is this doing?

Reply:
changed to passColor, its intent is to just assign color the selected color in multiple places at once